repair_follow_lot_location/models/stock_move.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2025 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+
+    _inherit = "stock.move"
+
+    def _action_done(self, cancel_backorder=False):
+        res = super()._action_done(cancel_backorder=cancel_backorder)
+        done_moves = self.filtered(lambda m: m.state == "done")
+        if not done_moves:
+            return res
+        lots = done_moves.move_line_ids.lot_id
+        if not lots:
+            return res
+        repair_orders = self.env["repair.order"].search(
+            [
+                ("follow_lot_location", "=", True),
+                ("lot_id", "in", lots.ids),
+                ("state", "not in", ["done", "cancel"]),
+            ]
+        )
+        for repair_order in repair_orders:
+            move = done_moves.filtered(
+                lambda m, lot=repair_order.lot_id: lot in m.move_line_ids.lot_id
+            )
+            repair_order.location_id = move.location_dest_id[:1]
```

Comment:
a loop into a loop.....
IMO after the search, create a dict of repair_order by lot_id.
Then in a single loop on moves, you can assign the right location...
```suggestion
        repair_by_lot = {r.lot_id: r for r in repair_orders}
        for move in moves:
            repair_order = repair_by_lot.get(move.lot_id)
            if repair_order:
                repair_order.location_id = move.location_dest_id[:1]
```

Reply:
I can't see how to avoid n2 loop since the lot is at move line level. please check now